lib/msal-common/src/cache/entities/AccountEntity.ts

```diff
@@ -170,10 +170,17 @@ export class AccountEntity {
             account.localAccountId = idToken?.claims?.oid || idToken?.claims?.sub || Constants.EMPTY_STRING;
 
             /*
-             * In B2C scenarios the emails claim is used instead of preferred_username and it is an array. In most cases it will contain a single email.
-             * This field should not be relied upon if a custom policy is configured to return more than 1 email.
+             * In B2C scenarios the emails claim is used instead of preferred_username and it is an array.
+             * In most cases it will contain a single email. This field should not be relied upon if a custom 
+             * policy is configured to return more than 1 email.
              */
-            account.username = idToken?.claims?.preferred_username || (idToken?.claims?.emails? idToken.claims.emails[0]: Constants.EMPTY_STRING);
+            account.username = 
+                idToken?.claims?.preferred_username ||
+                (
+                    idToken?.claims?.emails? 
```

Comment:
Spacing would help readability here

Reply:
Did some refactoring for readability, too many ternaries and optional changing, it was hard to tell that the precedence went preferred name > email > empty.